### PR TITLE
make some variable and function names more accurate in intermediate RNN tutorials

### PR DESCRIPTION
**Pull request by greaber**
Previously, processing a single training example was referred to as an "epoch".  I changed this to "step".  A downside of the change is that sometimes "step" is used to refer to a single step of the RNN in processing a training example.  But I don't think there will be too much confusion.

Previously, a function called "randomTrainingSet" in the generation tutorial returned the prepared tensors for a single training example.  I changed the name to "randomTrainingExample".

Previously, a function called "randomTrainingPair" in the classification tutorial returned the prepared tensors for a single training example.  I changed the name to "randomTrainingExample".  The old name also clashed with the use of the name "randomTrainingPair" in the generation tutorial.

A small remaining inconsistency is that both the classification and generation tutorials define functions called "train" but this function is invoked (after my name change) "train(*randomTrainingExample())" in the generation tutorial but cannot be invoked this way in the classification tutorial.

### CODE REVIEW

File: intermediate_source/char_rnn_classification_tutorial.py [hunk 1]
```diff
@@ -364,19 +364,19 @@ def timeSince(since):
 
 start = time.time()
 
-for epoch in range(1, n_epochs + 1):
```
Comment (spro) [inline intermediate_source/char_rnn_classification_tutorial.py]: `step` could get confusing considering we're using RNNs with multiple "time steps"... maybe `iter` and `n_iters`?

Comment (greaber) [inline intermediate_source/char_rnn_classification_tutorial.py]: OK, I can change it to be that way.  Also, I was thinking to delete the comment that says you don't understand why initializing the loss to zero works since I don't think it is good to give people more things to feel confused about, and anyway I just assumed pytorch allows you to create a Variable by adding a python constant to a Variable, which doesn't seem mysterious, or am I missing something?

Comment (spro) [inline intermediate_source/char_rnn_classification_tutorial.py]: Agreed

**Comment by chsasank** [comment]
@greaber Is this PR ready to merge?

**Comment by greaber** [comment]
Yes.

**Comment by chsasank** [comment]
Thanks a lot @greaber
